### PR TITLE
Add more charts to hawk/dove variable risk simulation

### PR DESCRIPTION
**Pull request by rlskoeser**
- chart rolling average percent hawk within each risk attitude
- chart number of agents for each risk attitude (current round only)

### CODE REVIEW

File: simulatingrisk/hawkdovevar/app.py [hunk 1]
```diff
@@ -32,10 +34,94 @@
     }
 )
 
+
+def agents_by_risk(model):
+    """plot total number of agents for each risk attitude"""
```
Comment (quadrismegistus) [inline simulatingrisk/hawkdovevar/app.py:39]: Maybe `plot_agents_by_risk` to be more explicit?

File: simulatingrisk/hawkdovevar/app.py [hunk 2]
```diff
@@ -32,10 +34,94 @@
...
+    agent_df = model.datacollector.get_agent_vars_dataframe().reset_index().dropna()
```
Comment (quadrismegistus) [inline simulatingrisk/hawkdovevar/app.py:40]: dropna is pretty aggressive, it'll drop any row with a single na. That's probably what you want but just as a note

File: simulatingrisk/hawkdovevar/app.py [hunk 3]
```diff
@@ -32,10 +34,94 @@
...
+                axis=alt.Axis(tickCount=model.num_neighbors + 1),
+                scale=alt.Scale(domain=[0, model.num_neighbors]),
+            ),
+            y=alt.Y("total", title="Number of agents"),
+        )
+    )
+    return solara.FigureAltair(bar_chart)
+
+
+def hawks_by_risk(model):
```
Comment (quadrismegistus) [inline simulatingrisk/hawkdovevar/app.py:73]: `plot_hawks_by_risk`?

File: simulatingrisk/hawkdovevar/app.py [hunk 4]
```diff
@@ -32,10 +34,94 @@
...
+    last_n_rounds = agent_df[agent_df.Step.gt(last_step - 60)].copy()
+    last_n_rounds["hawk"] = last_n_rounds.choice.apply(
+        lambda x: 1 if x == "hawk" else 0
+    )
+    # for each step and risk level, get number of agents and number of hawks
+    grouped = (
+        last_n_rounds.groupby(["Step", "risk_level"], as_index=False)
+        .agg(hawk=("hawk", "sum"), agents=("AgentID", "count"))
+        .sort_values("Step")
+    )
```
Comment (quadrismegistus) [inline simulatingrisk/hawkdovevar/app.py:92]: slick pandas chaining

File: simulatingrisk/hawkdovevar/app.py [hunk 5]
```diff
@@ -32,10 +34,94 @@
...
+        )
+    )
+    return solara.FigureAltair(chart)
+
+
 page = JupyterViz(
     HawkDoveVariableRiskModel,
     jupyterviz_params_var,
-    measures=[plot_hawks],
+    measures=[plot_hawks, agents_by_risk, hawks_by_risk],
```
Comment (quadrismegistus) [inline simulatingrisk/hawkdovevar/app.py:124]: `plot_*` would be consistent

**Review by quadrismegistus — APPROVED** [review]
Great work! Love the pandas and altair code

File: simulatingrisk/hawkdovevar/app.py (hunk 2 above)
Comment (rlskoeser) [reply simulatingrisk/hawkdovevar/app.py:40]: mesa seems to generate a few rows that are fully `na` in the first rounds before there is any data (possibly due to something I'm doing?); I don't think any of the data I'm generating is ever `na`; I suppose I could restrict this to a specific column like agent id, though.

File: simulatingrisk/hawkdovevar/app.py (hunk 5 above)
Comment (rlskoeser) [reply simulatingrisk/hawkdovevar/app.py:124]: yep, you're right - I was going for shorter titles but I'll relabel to make more explicit

**Pull request merged**